schematools/permissions.py

```diff
@@ -0,0 +1,107 @@
+from pg_grant import query, parse_acl_item
+from pg_grant.sql import grant, revoke
+from pg_grant import PgObjectType
+from .utils import to_snake_case
+from .importer import get_table_name
+from sqlalchemy.exc import SQLAlchemyError
+
+PUBLIC_SCOPE = "OPENBAAR"
+
+def introspect_permissions(engine, role):
+    schema_relation_infolist = query.get_all_table_acls(engine, schema='public')
+    for schema_relation_info in schema_relation_infolist:
+        if schema_relation_info.acl:
+            acl_list = [parse_acl_item(item) for item in schema_relation_info.acl]
+            for acl in acl_list:
+                if acl.grantee == role:
+                    print('role "{}" has priviliges {} on table "{}"'.format(role, ','.join(acl.privs), schema_relation_info.name))
+
+
+def revoke_permissions(engine, role):
+    grantee =role
+    schema_relation_infolist = query.get_all_table_acls(engine, schema='public')
+    for schema_relation_info in schema_relation_infolist:
+        if schema_relation_info.acl:
+            acl_list = [parse_acl_item(item) for item in schema_relation_info.acl]
+            for acl in acl_list:
+                if acl.grantee == role:
+                    print('revoking ALL priviliges of role "{}" on table "{}"'.format(role, schema_relation_info.name))
+                    revoke_statement = revoke("ALL", PgObjectType.TABLE, schema_relation_info.name, grantee)
+                    engine.execute(revoke_statement)
+
+def apply_schema_and_profile_permissions(engine, ams_schema, profiles, role, scope):
+    if ams_schema:
+        create_acl_from_schemas(engine, ams_schema, role, scope)
+    if profiles:
+        profile_list = profiles.values()
+        create_acl_from_profiles(engine, 'public', profile_list, role, scope)
+
+def create_acl_from_profiles(engine, schema, profile_list, role, scope):
+    acl_list = query.get_all_table_acls(engine, schema='public')
+    priviliges = ["SELECT", ]
+    grantee = role
+    for profile in profile_list:
+        if scope in profile["scopes"]:
+            for dataset, details in profile["schema_data"]["datasets"].items():
+                for item in acl_list:
+                    if item.name.startswith(dataset+"_"):
+                        grant_statement = grant(priviliges, PgObjectType.TABLE, item.name, grantee, grant_option=False, schema=schema)
+                        print(grant_statement)
+                        engine.execute(grant_statement)
+
+
+def create_acl_from_schema(engine, ams_schema, role, scope):
+    grantee = role
+    dataset_scope = ams_schema.auth if ams_schema.auth else {PUBLIC_SCOPE, }
+    dataset_scope_set = {dataset_scope} if isinstance(dataset_scope, str) else set(dataset_scope)
+
+    if len(dataset_scope_set) > 1 or PUBLIC_SCOPE not in dataset_scope:
```

Comment:
Zou de check op aanwezigheid dataset_scope_set ook als volgt kunnen doen?

`if dataset_scope_set or PUBLIC_SCOPE not in dataset_scope:`

Reply:
niet als hij op `len() > 1` test, maar weet niet waarom dat de bedoeling is?

Reply:
Het is hier de bedoeling te checken of er een scope anders dan PUBLIC_SCOPE in de set zit.
Mooier is denk ik : 
`if dataset_scope - PUBLIC_SCOPE:`